TAO/orbsvcs/orbsvcs/AV/AV_Core.cpp

```diff
@@ -1146,8 +1146,11 @@ TAO_AV_Core::get_control_flowname(const char *flowname)
   return flowname;
 }
 
-#if defined (ACE_HAS_EXPLICIT_STATIC_TEMPLATE_MEMBER_INSTANTIATION)
-template ACE_Singleton<TAO_AV_Core, ACE_Null_Mutex> *ACE_Singleton<TAO_AV_Core, ACE_Null_Mutex>::singleton_;
-#endif /* ACE_HAS_EXPLICIT_STATIC_TEMPLATE_MEMBER_INSTANTIATION */
 
 TAO_END_VERSIONED_NAMESPACE_DECL
+
+ACE_BEGIN_VERSIONED_NAMESPACE_DECL
+
+ACE_SINGLETON_TEMPLATE_INSTANTIATE(ACE_Singleton, TAO_AV_Core, ACE_Null_Mutex)
+
```

Comment:
With versioned namespaces enabled, the template ACE_Singleton belongs to the namespace ACE_x_y_z.  The explicit instantiation should be in that same namespace.  Looks like GCC also allows it to appear in the global namespace though I'm not sure this is technically legal C++.  I think the only instances of this we have are in tests or examples so I'm not too concerned but we could change them for consistency.